aws/data_source_aws_cloudtrail_service_account.go

```diff
@@ -0,0 +1,57 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// See http://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-supported-regions.html
+var cloudTrailServiceAccountPerRegionMap = map[string]string{
+	"us-east-1":      "086441151436",
+	"us-east-2":      "475085895292",
+	"us-west-1":      "388731089494",
+	"us-west-2":      "113285607260",
+	"ap-south-1":     "977081816279",
+	"ap-northeast-2": "492519147666",
+	"ap-southeast-1": "903692715234",
+	"ap-southeast-2": "284668455005",
+	"ap-northeast-1": "216624486486",
+	"ca-central-1":   "819402241893",
+	"eu-central-1":   "035351147821",
+	"eu-west-1":      "859597730677",
+	"eu-west-2":      "282025262664",
+	"sa-east-1":      "814480443879",
+}
+
+func dataSourceAwsCloudTrailServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsCloudTrailServiceAccountRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsCloudTrailServiceAccountRead(d *schema.ResourceData, meta interface{}) error {
+	region := meta.(*AWSClient).region
+	if v, ok := d.GetOk("region"); ok {
+		region = v.(string)
+	}
+
+	if accid, ok := cloudTrailServiceAccountPerRegionMap[region]; ok {
+		d.SetId(accid)
+		d.Set("arn", fmt.Sprintf("arn:%s:iam::%s:root", meta.(*AWSClient).partition, accid))
```

Comment:
Would probably be better to use https://github.com/terraform-providers/terraform-provider-aws/blob/master/vendor/github.com/aws/aws-sdk-go/aws/arn/arn.go#L28, thoughts?

Reply:
Agreed.
I added a new `arn.go` file and associated functions in https://github.com/terraform-providers/terraform-provider-aws/pull/1775 which I could use here if that's merged and I rebase.
Also mentioned in https://github.com/terraform-providers/terraform-provider-aws/issues/1783.

Reply:
Agreed, as it has been merged, should be all ok!